src/database/Database.cpp

```diff
@@ -443,6 +444,44 @@ Database::selectUpdateMap(
     return vecT.size();
 }
 
+template <typename T>
+void
+Database::decodeOpaqueXDR(std::string const& in, T& out)
```

Comment:
These should be free functions. I'm not even certain that this first overload should be defined in a "database" file; it's a generic utility function.

Reply:
Okay, I've made them into free functions.

I wasn't sure about putting the first overload in Database either, but I also wasn't sure where _else_ to put it.  Neither Decoder nor XDR knows about the other, and I figured they probably shouldn't, so that the new utility belonged in some third place.  And Database seemed like the place that already knew about decoding _and_ XDR, and indeed it was precisely the *SQL.cpp modules that needed decodeOpaqueXDR(), which I thought supported the idea of the functions belonging there.  But I'm open to suggestions for other places to move one or both of them to.

Reply:
I think it's fine for now. Right now we are only using it for the database. Maybe at a later time we'll have more clarity about it.